app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.test.js

```diff
@@ -7,7 +7,7 @@ import switchEthereumChain from './switch-ethereum-chain';
 const NON_INFURA_CHAIN_ID = '0x123456789';
 
 const mockRequestUserApproval = ({ requestData }) => {
-  return Promise.resolve(requestData);
+  return Promise.resolve(requestData.toNetworkConfiguration);
```

Comment:
Are there new `switchEthereumChain` scenarios we want to test with these changes?

Reply:
I know alot this will be covered by e2e tests we add after this PR but just wondering if there are any obvious unit tests to add. Non blocking question.

Reply:
good question, I haven't found any other than the tests i've fixed, but I didnt look too much deeper than just the ones that failed.